configurablecare/app/src/main/java/com/google/fhir/examples/configurablecare/screening/ListTasksViewModel.kt

```diff
@@ -0,0 +1,103 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.fhir.examples.configurablecare.screening
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.viewModelScope
+import ca.uhn.fhir.context.FhirContext
+import ca.uhn.fhir.context.FhirVersionEnum
+import com.google.android.fhir.search.Order
+import com.google.fhir.examples.configurablecare.FhirApplication
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import org.hl7.fhir.r4.model.Task
+import org.hl7.fhir.r4.model.Task.TaskStatus
+
+class ListScreeningsViewModel(application: Application) : AndroidViewModel(application) {
+
+  private val iParser = FhirContext.forCached(FhirVersionEnum.R4).newJsonParser()
+  private val taskManager =
+    FhirApplication.taskManager(getApplication<Application>().applicationContext)
+
+  val liveSearchedTasks = MutableLiveData<List<TaskItem>>()
+
+  fun getTasksForPatient(patientId: String, taskStatus: String) {
+    viewModelScope.launch {
+      liveSearchedTasks.value =
+        taskManager
+          .getTasksForPatient(
+            patientId = patientId,
+            extraFilter = { filter(Task.STATUS, { value = of(taskStatus) }) },
+            sort = { sort(Task.MODIFIED, Order.ASCENDING) }
+          )
+          .mapIndexed { index, fhirTask -> fhirTask.toTaskItem(index + 1) }
+    }
+  }
+
+  /**
+   * Alternatively we could cache the questionnaireStrings for each tasks in this viewModel instead
+   * of [runBlocking]
+   */
+  fun fetchQuestionnaireString(taskItem: TaskItem): String = runBlocking {
+    iParser.encodeResourceToString(
+      taskManager.fetchQuestionnaireFromTaskLogicalId(taskItem.resourceId)
+    )
+  }
+
+  data class TaskItem(
+    val id: String,
+    // for Task/123/... this should be 123
+    val resourceId: String,
+    val description: String,
+    val status: String,
+    val intent: String,
+    val dueDate: String,
+    val completedDate: String,
+    val owner: String,
+    val clickable: Boolean
+  ) {
+    override fun toString() = description
+  }
+}
+
+internal fun Task.toTaskItem(position: Int): ListScreeningsViewModel.TaskItem {
+  val taskResourceId = if (hasIdElement()) idElement.idPart else ""
+  val description = if (hasDescription()) description else "Sample Task Name"
+  // status and intent are always present
+  val taskStatus = status.toCode()
+  val taskIntent = intent.toCode()
+  val dueDate =
+    if (hasRestriction() && restriction.hasPeriod()) restriction.period.end.toString()
+    else "Sample End Date"
+  val completedDate = if (hasLastModified()) lastModified.toString() else dueDate
+  val owner = if (owner.hasDisplay()) owner.display else owner.reference
+  val clickable =
+    focus.reference.contains("Questionnaire") && taskStatus != TaskStatus.COMPLETED.name
```

Comment:
This should be:
`val clickable = focus.reference.contains("Questionnaire") && taskStatus != TaskStatus.COMPLETED.toCode()` because we are comparing with taskStatus (Line 82)

Reply:
Done.